### PR TITLE
parser: implement Restore for BinlogStmt, FlushStmt, KillStmt, DoStmt and AdminStmt

### PR DESCRIPTION
**Pull request by exialin**
### What problem does this PR solve? 
Implement `Restore` for `BinlogStmt`, `FlushStmt`, `KillStmt`.

Issue: https://github.com/pingcap/tidb/issues/8532

### CODE REVIEW

File: ast/misc.go [hunk 1]
```diff
@@ -847,7 +915,97 @@ type AdminStmt struct {
...
+	case AdminShowDDL:
+		ctx.WriteKeyWord("SHOW DDL")
+	case AdminShowDDLJobs:
+		ctx.WriteKeyWord("SHOW DDL JOBS")
+		if n.JobNumber != 0 {
+			ctx.WritePlainf(" %d", n.JobNumber)
+		}
+	case AdminShowNextRowID:
+		ctx.WriteKeyWord("SHOW ")
+		restoreTables()
```
Comment (kennytm) [inline ast/misc.go]: Check the error returned by `restoreTables()`

Comment (exialin) [reply ast/misc.go]: Already added.

**Comment by tiancaiamao** [comment]
LGTM

File: ast/misc.go [hunk 2]
```diff
@@ -426,7 +428,32 @@ type FlushStmt struct {
...
+				return errors.Annotatef(err, "An error occurred while restore FlushStmt.Tables[%d]", i)
+			}
+		}
+		if n.ReadLock {
+			ctx.WriteKeyWord(" WITH READ LOCK")
+		}
+	case FlushPrivileges:
+		ctx.WriteKeyWord("PRIVILEGES")
+	case FlushStatus:
+		ctx.WriteKeyWord("STATUS")
```
Comment (leoppro) [inline ast/misc.go]: add default as follows:
```
	default:
		return errors.New("Unsupported type of FlushTables")
```

Comment (exialin) [reply ast/misc.go]: Done.

**Pull request merged**